File: backend/src/main/java/com/petconect/backend/model/Product.java
package com.petconect.backend.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String description;
    private double price;
    private String imageUrl;
    private String measurementUnit;
    private Long ownerId;
    private String ownerName;
    private String ownerLocation;
    private String ownerPhone;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
} 

File: lib/core/models/product.dart
class Product {
  final int? id;
  final String name;
  final String description;
  final double price;
  final String imageUrl;
  final String measurementUnit;
  final int quantity;
  final String category;
  final int? ownerId;
  final String? ownerName;
  final String? ownerLocation;
  final String? ownerPhone;

  Product({
    this.id,
    required this.name,
    required this.description,
    required this.price,
    required this.imageUrl,
    required this.measurementUnit,
    required this.quantity,
    required this.category,
    this.ownerId,
    this.ownerName,
    this.ownerLocation,
    this.ownerPhone,
  });

  factory Product.fromJson(Map<String, dynamic> json) => Product(
    id: json['id'],
    name: json['name'] ?? '',
    description: json['description'] ?? '',
    price: (json['price'] ?? 0).toDouble(),
    imageUrl: json['imageUrl'] ?? '',
    measurementUnit: json['measurementUnit'] ?? '',
    quantity: json['quantity'] ?? 0,
    category: json['category'] ?? '',
    ownerId: json['ownerId'],
    ownerName: json['ownerName'],
    ownerLocation: json['ownerLocation'],
    ownerPhone: json['ownerPhone'],
  );

  Map<String, dynamic> toJson() => {
    'id': id,
    'name': name,
    'description': description,
    'price': price,
    'imageUrl': imageUrl,
    'measurementUnit': measurementUnit,
    'quantity': quantity,
    'category': category,
    'ownerId': ownerId,
    'ownerName': ownerName,
    'ownerLocation': ownerLocation,
    'ownerPhone': ownerPhone,
  };
} 